src/domain/session/SessionViewModel.ts

```diff
@@ -15,138 +15,162 @@ See the License for the specific language governing permissions and
-import {RoomGridViewModel} from "./RoomGridViewModel.js";
-import {SettingsViewModel} from "./settings/SettingsViewModel.js";
+import {LightboxViewModel} from "./room/LightboxViewModel";
+import {SessionStatusViewModel} from "./SessionStatusViewModel";
+import {RoomGridViewModel} from "./RoomGridViewModel";
+import {SettingsViewModel} from "./settings/SettingsViewModel";
 import {CreateRoomViewModel} from "./CreateRoomViewModel";
 import {ViewModel} from "../ViewModel";
-import {RoomViewModelObservable} from "./RoomViewModelObservable.js";
-import {RightPanelViewModel} from "./rightpanel/RightPanelViewModel.js";
+import {RoomViewModelObservable} from "./RoomViewModelObservable";
+import {RightPanelViewModel} from "./rightpanel/RightPanelViewModel";
+
+import type {Options as ViewModelOptions} from "../ViewModel";
+import type {Client} from "../../matrix/Client";
+import type {Room} from "../../matrix/room/Room";
+import type {IGridItemViewModel} from "./room/IGridItemViewModel";
+
+type Options = {
+    client: Client
+} & ViewModelOptions;
 
 export class SessionViewModel extends ViewModel {
-    constructor(options) {
+    client: Client;
+    private _sessionStatusViewModel: SessionStatusViewModel;
+    private _leftPanelViewModel: LeftPanelViewModel;
+    private _settingsViewModel?: SettingsViewModel;
+    private _roomViewModelObservable?: RoomViewModelObservable;
+    private _gridViewModel?: RoomGridViewModel;
+    private _createRoomViewModel?: CreateRoomViewModel;
+    private _rightPanelViewModel?: RightPanelViewModel;
+    private _lightboxViewModel?: LightboxViewModel;
+
+    constructor(options: Options) {
         super(options);
         const {client} = options;
-        this._client = this.track(client);
+        this.client = this.track(client);
         this._sessionStatusViewModel = this.track(new SessionStatusViewModel(this.childOptions({
             sync: client.sync,
             reconnector: client.reconnector,
             session: client.session,
         })));
-        this._leftPanelViewModel = this.track(new LeftPanelViewModel(this.childOptions({session: this._client.session})));
-        this._settingsViewModel = null;
-        this._roomViewModelObservable = null;
-        this._gridViewModel = null;
-        this._createRoomViewModel = null;
+        this._leftPanelViewModel = this.track(new LeftPanelViewModel(this.childOptions({session: this.client.session})));
         this._setupNavigation();
     }
 
-    _setupNavigation() {
+    _setupNavigation(): void {
         const gridRooms = this.navigation.observe("rooms");
         // this gives us a set of room ids in the grid
-        this.track(gridRooms.subscribe(roomIds => {
+        this.track(gridRooms.subscribe((roomIds: string[] | undefined) => {
             this._updateGrid(roomIds);
         }));
         if (gridRooms.get()) {
-            this._updateGrid(gridRooms.get());
+            this._updateGrid(gridRooms.get() as string[] | undefined);
```

Comment:
`gridRooms` is a `SegmentObservable<SegmentType>`. _We_ know that `gridRooms.get()` will give us a `string[] | undefined`, because [`SegmentType['rooms']` is a `string[]`](https://github.com/ibeckermayer/hydrogen-web/blob/2031e4b4b8aff1fd5aa6ba3a718eb60ce1c8233b/src/domain/navigation/index.ts#L27), however typescript isn't able to figure that out given how `SegmentObservable` is written. I couldn't find any good way to get typescript to understand this automatically, so we give it some help here.

This was [my best attempt](https://github.com/ibeckermayer/hydrogen-web/commit/2031e4b4b8aff1fd5aa6ba3a718eb60ce1c8233b), but I think that letting `R` become anything is too loose of a definition, and that its better to "cast" things inline as needed.